mpf/devices/multiball.py

```diff
@@ -237,15 +237,35 @@ def _ball_drain_count_balls(self, balls, **kwargs):
         desc: The multiball called (name) has lost a ball after ball save expired.
         '''
 
-        if not self.machine.game or self.machine.game.balls_in_play - balls < 1:
-            self.balls_added_live = 0
-            self.balls_live_target = 0
-            self.machine.events.remove_handler(self._ball_drain_count_balls)
-            self.machine.events.post("multiball_{}_ended".format(self.name))
-            '''event: multiball_(name)_ended
-            desc: The multiball called (name) has just ended.
-            '''
-            self.debug_log("Ball drained. MB ended.")
+        restart_grace_period_ms = self.config['restart_grace_period'].evaluate([])
+
+        if not self.machine.game or (self.machine.game.balls_in_play - balls < 1 and restart_grace_period_ms==0):
+            self._multiball_end()
+        elif restart_grace_period_ms>0:
+            if self.machine.game.balls_in_play == 0:
+                self.delay.run_now(f"{self.name}_restart_grace_period")
```

Comment:
Please remove `{self.name}` from the delay name, it's a computational cost to generate the formatted string and unnecessary because `self.delay` is isolated in scope to just this particular instance of multiball.